packages/driver/src/cypress/commands.ts

```diff
@@ -173,6 +171,26 @@ export default {
         queries[name] = fn
         cy.addQuery({ name, fn })
       },
+
+      overwriteQuery (name: string, fn: () => QueryFunction) {
```

Comment:
Why wouldn't this logic be 1:1 with add command beyond the error handling and using `addQuery`?

Reply:
The main difference is that queries don't have the concept of "parent/child" that non-queries use.

The call signature of a parent command is `fn(...args)` (not accepting a subject at all), while a child command is `fn(subject, ...args)` (taking the subject as the first argument).

All queries use the signature of `fn(...args)`, so we get to skip the conditional check of slicing the subject off of `args` - it's never there, and the args require no manipulation.

----

The other difference that the `commands` object stores `commands[name] = { name, fn, type, prevSubject }` while `queries` stores only `queries[name] = fn`.

This is because when a command is overwritten, it needs to maintain the same `type` and `previousSubject` values as the original; queries have neither of these, and all we need to create the overwrite is the original function.